### ISSUE TITLE
Running CLI in directory without write permission

### ISSUE THREAD

**Issue by mbadams5** [NREL/EnergyPlus#5062]
When I run EnergyPlus through the CLI from the /Applications/EnergyPlus-8-3-0 folder it hangs on "Processing Input File". 

This is due to my user login not having write permission to that folder. If I run the same command with 'sudo' then it works as expected. 

```
computer_name:EnergyPlus-8-3-0 user_name$ ./energyplus -a -r -d ~/Desktop/Outputs/ -w Weathe
rData/USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw ExampleFiles/RefBldgMediumOf
ficeNew2004_Chicago.idf 
EnergyPlus Starting
EnergyPlus, Version 8.3.0-6d97d074ea, YMD=2015.07.31 14:05
Processing Data Dictionary
Processing Input File
^C
computer_name:EnergyPlus-8-3-0 user_name$ 
```

I think InputProcessor is writing to some temporary file during its processing, which is trying to write to the working directory instead of being put in the -d ~/Desktop/Outputs folder like all the rest of the 'normal' outputs. 

If I have write permission to the output folder then that should be where all outputs are placed, including temporary outputs.

fyi @kbenne @Myoldmopar @DeadParrot

**Comment by DeadParrot** [comment]
@mbadams5 If it is a scratch file created in the Fortran emulation of ObjexxFCL then it doesn't know about EnergyPlus output directories and has to use its own logic to find a writable location. I'm happy to look at that logic and improve it. Can you tell me where in the code it is trying to create the file and what the directory situation is (current, writable, etc.).

**Comment by mbadams5** [comment]
@DeadParrot I don't have an exact location yet, but that is my (informed) guess. I don't have time to track it down so i'll let @nealkruis do that.

**Comment by EnergyArchmage** [comment]
I would guess it is the "eplusout.audit" file that InputProcessor is writing to.

**Comment by DeadParrot** [comment]
@mbadams5 so are you running from a current directory that isn't writable to you?

**Comment by mbadams5** [comment]
@DeadParrot correct

**Comment by mbadams5** [comment]
@DeadParrot [Look here](https://github.com/NREL/EnergyPlus/blob/3ed10521c308b40c10bdb45007e9d9573b39f10f/src/EnergyPlus/SimulationManager.cc#L2486), I quickly ran a debug build in a folder without write permission in LLDB and this is where it is (currently) getting hung up.

**Comment by DeadParrot** [comment]
@mbadams5 OK, thanks. This isn't a scratch file and it looks like the eplusout.iperr is something new and the i/o may not be set up quite right. First, if you wrote it to the current directory it during the run why can't you then open it there. Or does it not exist yet when you are trying to open it? It looks like eplusout.iperr may be placed in the output directory instead of locally but this open is wrong: the open is using the raw file name (assuming local directory) instead of `outputIperrFileName`.

Probably not relevant to this but for reference sake, to read a file from a non-writable directory you would pass a read-only flags argument to the `open` like:

``` cpp
IOFlags flags; flags.readonly_on();
gio::open( CacheIPErrorFile, "eplusout.iperr", flags );
```

In short, I don't think this is an ObjexxFCL issue but I'm willing to be proven wrong.

**Comment by nealkruis** [comment]
I think @DeadParrot is right. Somehow this one `gio::open` statement escaped the refactor of raw file names into the new string variables.

**Comment by mbadams5** [comment]
@nealkruis @Myoldmopar @kbenne fyi

https://unmethours.com/question/9920/opening-idf-generated-by-os-using-energyplus/?answer=9929#post-id-9929
